pkg/controller/direct/sql/sqlinstance_equality.go

```diff
@@ -22,59 +22,52 @@ import (
 	api "google.golang.org/api/sqladmin/v1beta4"
 )
 
-func InstancesMatch(desired *api.DatabaseInstance, actual *api.DatabaseInstance, diff *structuredreporting.Diff) bool {
+func DiffInstances(desired *api.DatabaseInstance, actual *api.DatabaseInstance) *structuredreporting.Diff {
+	diff := &structuredreporting.Diff{}
 	if desired == nil && actual == nil {
-		return true
+		return diff
 	}
 	if !PointersMatch(desired, actual) {
-		return false
+		return diff
```

Comment:
PointersMatch is a terrible name for that method. At this point we know that exactly one of desired/actual are not nil. Returning an empty diff does not seem correct. Also suspicious this does not conform to the old behavior.